src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooService.java

```diff
@@ -566,11 +570,17 @@ public Result onBuildCompleted(ProgrammingExerciseParticipation participation, O
             // Do not remove this save, otherwise Hibernate will throw an order column index null exception on saving the build logs
             programmingSubmission = programmingSubmissionRepository.save(programmingSubmission);
 
+            // todo wip, save currently necessary, because of relationship between result and submission
```

Comment:
can we remove it and instead document why saving the result here is necessary?

Reply:
adjusted in: ed4e902